src/udev.c

```diff
@@ -224,6 +235,119 @@ udev_find_more_about_optical(struct udev_device *udev_device,  device_t *device,
   udev_enumerate_unref(enumerate);
 }
 
+static int
+is_ethernet_interface(struct usb_interface_descriptor *interface)
+{
+  return (interface->bInterfaceClass == COMMUNICATIONS_CLASS &&
+    interface->bInterfaceSubClass == ETHERNET_NETWORKING_SUBCLASS);
+}
+
+static int
+is_bluetooth_interface(struct usb_interface_descriptor *interface)
+{
+  return (interface->bInterfaceClass == WIRELESS_CLASS &&
+    interface->bInterfaceSubClass == RADIO_FREQUENCY_SUBCLASS &&
+    interface->bInterfaceProtocol == BLUETOOTH_PROTOCOL);
+}
+
+/**
+ * Identify vendor-specific interfaces with endpoints that support
+ * bulk data transfers both in and out with a packet size of at least
+ * TYPICAL_NIC_PACKET_SIZE (512). Though all the USB NICs that were
+ * tested had such an endpoint and none of the other USB devices did,
```

Comment:
Doesn't this packet size test also run the risk of including external USB disks?